docs/HOWTO/20_Puppet/Hiera_eyaml.rst

```diff
@@ -0,0 +1,194 @@
+====================================
+
+This section steps through configuring the :term:`hiera-eyaml`  backend
+for a puppet environment.  This backend allows you to encrypt data in
+the hiera files.
+
+An example of how to use the :command:`eyaml` command provided by the :term:`hiera-eyaml`
+gem to encrypt data is included at the end.
+
+More details on configuring and using eyaml can be found in the puppet
+documents for `Configuring hiera-eyaml`_ and in the `hiera-eyaml documentation`_.
+
+Install the gem.
+----------------
+
+The hiera-eyaml backend has been included in Puppet Server since version 5.2.0.
+If the Puppet Server does not already have the hiera-eyaml gem installed run:
+
+.. code-block::  bash
+
+   $ puppetserver gem install hiera-eyaml
+
+The hiera-eyaml gem can be installed by users into their ruby environment. This
+will provide the :command:`eyaml` command used to encrypt the data.
+
+.. code-block::  bash
+
+   $ gem install hiera-eyaml
+
+Set up the hiera eyaml hierachy
+-------------------------------
+
+On the Puppet Server, in the top level of the puppet environment,
+:file:`/etc/puppetlabs/code/environments/<environment name>`, edit the
+:file:`hiera.yaml` file and add the eyaml hierarchy.
+
+The following is an example of an eyaml hierarchy.  It was added to the
+:code:`simp-environment-skeleton` module in the  :file:`hiera.yaml` file in
+SIMP 6.6.0.  This example is a simple one.  You can have several entries
+that use different keys.
+
+Note the :code:`lookup_key` and :code:`options` keys.  Also note
+that the files have a :code:`.eyaml` extension.
+
+.. code-block:: yaml
+
+  ---
+  defaults:
+    - datadir: data
+
+  hierarchy:
+    - name: Eyaml Data
+      lookup_key: eyaml_lookup_key
+      paths:
+      - "secrets/hosts/%{trusted.certname}.eyaml"
+      - "secrets/hosts/%{facts.fqdn}.eyaml"
+      - "secrets/hosts/%{facts.hostname}.eyaml"
+      - "secrets/domains/%{facts.domain}.eyaml"
+      - "secrets/hostgroups/%{::hostgroup}.eyaml"
+      - "secrets/common.eyaml"
+      # The keys list below needs to be generated using eyaml gem.
+      # See simp docs for details.
+      options:
+        pkcs7_private_key: /var/lib/puppet/keys/private_key.pkcs7.pem
+        pkcs7_public_key: /var/lib/puppet/keys/public_key.pkcs7.pem
+  ...
+
+Generate the keys
+-----------------
+
+The keys location is configured under :code:`options` in the above hierarchy.
+Use the :command:`eyaml` command to generate these keys. Make sure the permissions on the
+keys are set securely and the Puppet Server has access to them.
+
+.. code-block:: bash
+
+  $ mkdir -p /var/lib/puppet/keys/
+  $ sudo /opt/puppetlabs/puppet/bin/eyaml  createkeys \
+     --pkcs7-private-key=/var/lib/puppet/keys/private_key.pkcs7.pem \
+     --pkcs7-public-key=/var/lib/puppet/keys/public_key.pkcs7.pem
+  $ cd /var/lib/
+  $ sudo chown -R puppet puppet
+  $ sudo chmod  500 puppet
+  $ sudo chmod  400 puppet/*.pem
+
+Make sure the keys are backed up.
```

Comment:
Do you want this to be in an IMPORTANT block?

Reply:
No, I find those blocks really annoying when looking at the documentation.  They should know to backup their stuff.  It is just a reminder.